cluster-api/src/main/java/com/sequenceiq/cloudbreak/cluster/api/ClusterCommissionService.java

```diff
@@ -13,4 +14,6 @@
     Map<String, InstanceMetaData> collectHostsToCommission(@Nonnull HostGroup hostGroup, Set<String> hostNames);
 
     Set<String> recommissionClusterNodes(Map<String, InstanceMetaData> hostsToRecommission);
+
+    void recommissionHosts(List<String> hostsToRecommission);
```

Comment:
Any chance this could make use of the existing recommissionClusterNodes(Map<String, InstanceMetadata>) - here and the other affected classes.

Reply:
I do not have InstanceMetadata, hence started a new interface. The code is mostly re-used across the two.